* Don't include fcntl.h on Solaris, or you get compilation errors since
  kernel views are enabled.  We don't seem to need it, so this is the
  easiest fix.

--- gdb/defs.h
+++ gdb/defs.h
@@ -37,7 +37,9 @@
 #include <unistd.h>
 #endif
 
+#ifndef __sun__
 #include <fcntl.h>
+#endif
 
 /* First include ansidecl.h so we can use the various macro definitions
    here and in all subsequent file inclusions.  */
